### PR TITLE
Allow for sending data

### PR DESCRIPTION
**Pull request by LVala**
Additionally, modify the examples to create a media loop (browser -> Elixir PeerConnection -> browser).

### CODE REVIEW

File: lib/ex_webrtc/dtls_transport.ex [hunk 1]
```diff
@@ -121,18 +127,25 @@ defmodule ExWebRTC.DTLSTransport do
...
     {:noreply, state}
   end
 
   @impl true
-  def handle_cast({:send_data, _data}, state) do
-    Logger.error(
-      "Attempted to send data when DTLS handshake was not finished or ICE Transport is unavailable"
-    )
+  def handle_cast({:send_rtp, _data}, state) do
+    Logger.warning("Attemped to send RTP before DTLS handshake has been finished")
```
Comment (mickel8) [inline lib/ex_webrtc/dtls_transport.ex:142]: Let's explicitly state that we ignore the packet. This way we will always be sure reading docs that this behaviour doesn't have any impact on the PC.

```suggestion
    Logger.warning("Attemped to send RTP before DTLS handshake has been finished. Ignoring.")
```

File: lib/ex_webrtc/dtls_transport.ex [hunk 2]
```diff
@@ -195,16 +209,16 @@ defmodule ExWebRTC.DTLSTransport do
         state = %{state | buffered_packets: packets}
         update_dtls_state(state, :connecting)
 
-      {:handshake_finished, _, remote_keying_material, profile, packets} ->
+      {:handshake_finished, lkm, rkm, profile, packets} ->
         Logger.debug("DTLS handshake finished")
         ICEAgent.send_data(state.ice_agent, packets)
         # TODO: validate fingerprint
-        state = setup_srtp(state, remote_keying_material, profile)
+        state = setup_srtp(state, lkm, rkm, profile)
```
Comment (mickel8) [inline lib/ex_webrtc/dtls_transport.ex:216]: we could make setup_srtp return :ok

File: lib/ex_webrtc/peer_connection.ex [hunk 3]
```diff
@@ -326,26 +337,56 @@ defmodule ExWebRTC.PeerConnection do
...
+  def handle_cast({:send_rtp, track_id, packet}, state) do
+    sdes_id =
+      Enum.find_value(state.demuxer.extensions, fn
+        {ext_id, {ExRTP.Packet.Extension.SourceDescription, :mid}} -> ext_id
+        _ -> nil
+      end)
+
+    # TODO: iterating over transceivers is not optimal
+    # but this is, most likely, going to be refactored anyways
+    maybe_transceiver =
```
Comment (mickel8) [inline lib/ex_webrtc/peer_connection.ex:364]:
```suggestion
    transceiver =
```

File: lib/ex_webrtc/peer_connection.ex [hunk 4]
```diff
@@ -377,11 +418,16 @@ defmodule ExWebRTC.PeerConnection do
...
 
   @impl true
-  def handle_info({:dtls_transport, _pid, {:rtp_data, data}}, state) do
-    case Demuxer.demux(state.demuxer, data) do
-      {:ok, demuxer, mid, packet} ->
-        notify(state.owner, {:data, {mid, packet}})
-        {:noreply, %__MODULE__{state | demuxer: demuxer}}
+  def handle_info({:dtls_transport, _pid, {:rtp, data}}, state) do
+    with {:ok, demuxer, mid, packet} <- Demuxer.demux(state.demuxer, data),
+         %RTPTransceiver{} = t <- Enum.find(state.transceivers, &(&1.mid == mid)) do
```
Comment (mickel8) [inline lib/ex_webrtc/peer_connection.ex:423]: Not sure about this one. I belive this should be in the `do end` body as this should always be true

File: lib/ex_webrtc/dtls_transport.ex (hunk 2 above)
Comment (LVala) [reply lib/ex_webrtc/dtls_transport.ex:216]: why tho? Do you mean that `setup_srtp` could fail when provided with these keying materials or profile?

File: lib/ex_webrtc/peer_connection.ex (hunk 4 above)
Comment (LVala) [reply lib/ex_webrtc/peer_connection.ex:423]: Yeah, but atm demuxer does not check that, I could change this in this pull request, but the whole receiving/sending data business is quite temporary anyways, isn't it?

**Pull request merged**